nifi-nar-bundles/nifi-ocr-bundle/nifi-ocr-processors/src/main/java/org/apache/nifi/processors/ocr/TesseractOCRProcessor.java

```diff
@@ -0,0 +1,361 @@
+            .name("failure")
+            .description("Failed to attempt OCR on input image")
+            .build();
+
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(TESS_DATA_PATH);
+        descriptors.add(TESSERACT_LANGUAGE);
+        descriptors.add(TESSERACT_PAGE_SEG_MODE);
+        descriptors.add(TESSERACT_CONFIGS);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_UNSUPPORTED_IMAGE_FORMAT);
+        relationships.add(REL_ORIGINAL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+
+        List<PropertyDescriptor> descriptorsNew = new ArrayList<>();
+
+        descriptorsNew.add(TESS_DATA_PATH);
+        descriptorsNew.add(new PropertyDescriptor.Builder()
+                .fromPropertyDescriptor(TESSERACT_LANGUAGE)
+                .allowableValues(SUPPORTED_LANGUAGES)
+                .build());
+        descriptorsNew.add(TESSERACT_PAGE_SEG_MODE);
+        descriptorsNew.add(TESSERACT_CONFIGS);
+
+        return descriptorsNew;
+    }
+
+    @Override
+    public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
+        super.onPropertyModified(descriptor, oldValue, newValue);
+
+        if (descriptor.equals(TESS_DATA_PATH)) {
+            getLogger().debug("Tesseract Install path was changed. Building list of supported languages");
+            SUPPORTED_LANGUAGES.clear();
+            SUPPORTED_LANGUAGES.add("eng");
+            //File will always exist since the Validator will take care of that.
+            File[] files = getTesseractLanguages(newValue);
+
+            //Guard against creating an empty list of allowable values in case the user points to an invalid directory
+            if (files != null && files.length > 0) {
+                for (int i = 0; i < files.length; i++) {
+                    getLogger().debug("Found Tesseract supported language: " + files[i].getName());
+                    SUPPORTED_LANGUAGES.add(StringUtils.split(files[i].getName(), ".")[0]);
+                }
+            } else {
+                getLogger().debug("No languages found in user specified Tessdata directory: '" + newValue + "'");
+            }
+
+        }
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        //Setup the Tesseract instance once the processor is scheuled
+        tessInstance = new Tesseract();
```

Comment:
Not to be a pain, but i keep stumbling on the name of the variable. I constantly think its a spelling error ;) While I'll leave it up to you, IMHO naming it simply 'tesseract' would be better 
```
tesseract = new Tesseract();
```